### PR TITLE
fix: continue update process in middle of update if spec.replicas is 0

### LINKED ISSUE #1690: Manifest update in middle of canary is not deploying the latest
## Summary 

1. New image triggers canary and is in steps
2. while canary is paused, there is another image update
3. Should start new canary deployment but it is not happening

workaround: delete old RS and update

## Diagnostics

What version of Argo Rollouts are you running?
v1.1.0
```
# Paste the logs from the rollout controller

# Logs for the entire controller:
kubectl logs -n argo-rollouts deployment/argo-rollouts

# Logs for a specific rollout:
kubectl logs -n argo-rollouts deployment/argo-rollouts | grep rollout=<ROLLOUTNAME>
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

### PR DESCRIPTION
**Pull request by huikang**
- if canary.spec.replicas==0 as in the canary steps, update continues

Signed-off-by: Hui Kang <hui.kang@salesforce.com>

fix #1690 

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by harikrongali** [comment]
thanks for fixing the issue. @huikang

File: utils/replicaset/replicaset.go [hunk 1]
```diff
@@ -630,5 +630,5 @@ func IsReplicaSetReady(rs *appsv1.ReplicaSet) bool {
 	}
 	replicas := rs.Spec.Replicas
 	readyReplicas := rs.Status.ReadyReplicas
-	return replicas != nil && *replicas != 0 && readyReplicas != 0 && *replicas <= readyReplicas
+	return (replicas != nil && *replicas != 0 && readyReplicas != 0 && *replicas <= readyReplicas) || (replicas != nil && *replicas == 0)
```
Comment (harikrongali) [inline utils/replicaset/replicaset.go:633]: can you update description to the function with added condition

Comment (harikrongali) [inline utils/replicaset/replicaset.go:633]: can you please add the given scenario as an e2e test?

**Review by jessesuen — APPROVED** [review]

**Comment by jessesuen** [comment]
Great job!

**Pull request merged**